docs/standards/generic-standards/lsp1-universal-receiver.md

```diff
@@ -35,26 +45,50 @@ This standard defines a single function named `universalReceiver(...)` that coul
+> Receiving contracts should consider the `typeId` parameter to **decode the data correctly**.
+
+The `universalReceiver(...)` function **emits an event with the data passed to it and some additional data**. 
+
+![universalReceiver function emits an event](/img/standards/lsp1/universal-receiver-event.jpeg)
+
+For example, **to mark the receiving of a specific token**, during a token transfer, the token contract can call the `universalReceiver(..)` function of the token recipient with:
+
+- bytes32 `typeId`: **Hash**('ERCXXXXTokenReceived')
+- bytes `data`: **packedData**(amount of token sent, the sender address, the block timestamp)
+
+In this way, instead of **listening to all the events of the token contrats on the network**, and checking which one of these transfers is relative to the recipient, users can listen to the **[UniversalReceiver](../smart-contracts/lsp0-erc725-account.md#universalreceiver-1)** event on the contract implementing the `universalReceiver(..)` and know the token transfer details.  
+
+
+Also, the function can then implement **custom logic** to make the contract behave differently based on the data received. For instance, the `universalReceiver(...)` function, if customized, can do the following:
 
 - Reverting on calls to completely disallow the smart contract from receiving assets, information, etc. :x:
 - Registering the received assets inside the contract storage (see [LSP5 - Received Assets](../universal-profile/lsp5-received-assets.md)). :heavy_plus_sign:
+- Forwarding all the received assets to an external vault or a staking contract.
+
+![universalReceiver function execute custom logic](/img/standards/lsp1/universal-receiver-logic.jpeg)
 
-> Receiving contracts should consider the `typeId` parameter to **decode the data correctly**.
 
-![schema of universal receiver transaction](/img/standards/lsp1/ur-transaction.jpeg)
 
 ## Extension
 
 :::info
 
-See the **[LSP1-UniversalReceiverDelegate](../universal-profile/lsp1-universal-receiver-delegate.md)** standard for more details.
+See the **[LSP1-UniversalReceiverDelegate](../generic-standards/lsp1-universal-receiver-delegate.md)** standard for more details.
 
 :::
 
-LSP1-UniversalReceiverDelegate is an **optional extension** to the LSP1-UniversalReceiver Standard. As well as notifying a contract about the incoming and outgoing transactions via an event, it will delegate the `universalReceiver(...)` functionality to an external contract that can **handle and react to specific calls** with its custom logic.
+Customizing the `universalReceiver(..)` function is an option that users have to allow the function to **behave differently on different actions**. However, it's not advised to hardcode the logic of reacting on specific actions inside the function, as **this logic may need to change in the future** depending on several factors (eg. the vault where the tokens are forwarded get hacked, new staking contract is deployed, decided to revert on specific tokens later). 
```

Comment:
_Suggested changes_

```suggestion
Overriding and customizing the `universalReceiver(..)` function is an option for users to allow **different behaviours depending on the data received**. However, it's not advised to hardcode the logic of reacting to specific actions inside the function because **this logic may need to change in the future** depending on several factors (eg. the vault where the tokens are forwarded gets compromised, a new staking contract is deployed, decided to revert on specific tokens later). 
```